src/devices/cpu/tlcs90/tlcs90.cpp

```diff
@@ -1422,18 +1431,14 @@ void tlcs90_device::execute_set_input(int inputnum, int state)
 
 void tlcs90_device::set_irq_line(int irq, int state)
 {
-	if ( ((m_irq_line_state >> irq)&1) == state ) return;
-
```

Comment:
What’s the reason for removing this check?  It looks like it would generate a spurious transition if something calls `set_input_line` with the same values twice in a row.

Reply:
This was done because some of the IRQ of the device are edge-triggered and are automatically cleared as soon as they are acknowledged. 
The code in tlcs90.cpp didn't reset the state of the irq lines (`m_irq_line_state`) when clearing it in that scenario and prevents subsequent (not at the same time) interrupts to be acknowledged, so, I simply removed those checks.
However, on a second look, a better solution would be maintain them and reset `m_irq_line_state` on the `take_interrupt` function, at the same point the irq is cleared.

Reply:
> This was done because some of the IRQ of the device are edge-triggered and are automatically cleared as soon as they are acknowledged.

You aren’t understanding what the code is doing.  `m_irq_line_state` holds the level of the interrupt lines so that it can detect an edge.  If it doesn’t track the current state of the line, it won’t know if it’s seeing an edge or the same value again.

> The code in tlcs90.cpp didn't reset the state of the irq lines (`m_irq_line_state`) when clearing it in that scenario and prevents subsequent (not at the same time) interrupts to be acknowledged, so, I simply removed those checks.

Well yes, the state of the signal on an interrupt input doesn’t magically change when the CPU services the interrupt.  It will only change when the circuit connected to the input changes it.

The interrupt pin must be lowered and raised again before a subsequent edge-triggered interrupt will be detected.  That’s the correct behaviour.  You’ve broken that.

> However, on a second look, a better solution would be maintain them and reset `m_irq_line_state` on the `take_interrupt` function, at the same place the irq is cleared there.

That’s no better.  It’s still magically changing the state of an input that isn’t generated by the CPU.  The code for whatever is attached to the pin is responsible for lowering the line before raising it again to cause another edge-triggered interrupt.